tests/integration/index/query_with_composite_index_only_filter_test.go

```diff
@@ -0,0 +1,767 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+const schemaWithNameAgeIndex = `
+	type User @index(fields: ["name", "age"]) {
+		name: String
+		age: Int
+		email: String
+	}`
+
+const schemaWithAgeNameIndex = `
+	type User @index(fields: ["age", "name"]) {
+		name: String
+		age: Int
+		email: String
+	}`
+
+func TestQueryWithCompositeIndex_WithEqualFilter_ShouldFetch(t *testing.T) {
+	req1 := `query {
+		User(filter: {name: {_eq: "Islam"}}) {
+			name
+			age
+		}
+	}`
+	req2 := `query {
+		User(filter: {name: {_eq: "Islam"}, age: {_eq: 32}}) {
+			name
+			age
+		}
+	}`
+	req3 := `query {
+		User(filter: {name: {_eq: "Islam"}, age: {_eq: 66}}) {
+			name
+			age
+		}
+	}`
+	test := testUtils.TestCase{
+		Description: "Test filtering on composite index with _eq filter",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: schemaWithNameAgeIndex,
```

Comment:
suggestion: I do feel like you are hiding the most important part of this test here.  The index is what you are testing, and as reader, what I am most interesting in seeing, and yet it is not visible within the test.

Reply:
done